CollectNewspaperKarel.java

```diff
@@ -1,17 +1,38 @@
-/*
- * File: CollectNewspaperKarel.java
- * --------------------------------
- * At present, the CollectNewspaperKarel subclass does nothing.
- * Your job in the assignment is to add the necessary code to
- * instruct Karel to walk to the door of its house, pick up the
- * newspaper (represented by a beeper, of course), and then return
- * to its initial position in the upper left corner of the house.
- */
-
-import stanford.karel.*;
-
-public class CollectNewspaperKarel extends SuperKarel {
-
-	// You fill in this part
-
-}
+/*
+ * File: CollectNewspaperKarel.java
+ * --------------------------------
+ * At present, the CollectNewspaperKarel subclass does nothing.
+ * Your job in the assignment is to add the necessary code to
+ * instruct Karel to walk to the door of its house, pick up the
+ * newspaper (represented by a beeper, of course), and then return
+ * to its initial position in the upper left corner of the house.
+ */
+
+import stanford.karel.*;
+
+public class CollectNewspaperKarel extends SuperKarel {
+
+	// You fill in this part
+
+	public void run() {
+		move();
+		move();
+		turnRight();
+		move();
+		turnLeft();
+		move();
+		pickBeeper();
+		turnAround();
+		move();
+		turnRight();
+		move();
+		turnLeft();
+		move();
+		move();
+		turnAround();

```

Comment:
Your code works but next time you should learn to break down your problem into simpler parts that can be reflected in code. For example for this question one way to approach it would be to break it as follows:

```
public void run() {
    pickUpNewsPaper();
    returnToInitialPosition();
}
```